### PR TITLE
Make tags sortable with drag&drop

### PR DESCRIPTION
**Pull request by AdrienLemaire**
Feature that enables sorting of tags by drag & drop.

To merge with [HabitRPG-shared/#276](https://github.com/HabitRPG/habitrpg-shared/pull/276).

### CODE REVIEW

**Comment by lefnire** [comment]
I'm getting some weird behavior here.
1. `major` Often after sorting, a ghost tag is created. Drop a tag into place, and suddenly a blank tag appears at the end. I haven't pinned it, but it seems usually after you drop the tag at the end of the list. Sometimes the ghost tag doesn't appear until after you refresh.
2. `minor` When sorting tags, it's quite difficult to sort them (often, dragging doesn't do anything - you have to scrub it into place until it triggers). This may be something we can't change, likely just a jQuery UI issue. Alternatively we can look into replacing our custom sortable directive with https://github.com/angular-ui/ui-sortable (incl. for task sorting, etc). If this is a big ol' rabbit hole, ignore this request.

**Comment by lefnire** [comment]
longpress

**Comment by AdrienLemaire** [comment]
I knew about 2, but I thought it was still better than nothing. Not sure how to improve it.

I didn't notice about 1. though :S That's really annoying. Same thing, not sure how to fix, and I don't have time to look into it now, sorry :(

**Comment by AdrienLemaire** [comment]
@lefnire and looks like you're the expert :p 
I used your method **habitrpgSortable** from `public/js/directives/directives.js` to write **tagsSortable**. 

Well, I'll try to debug this whenever I get time, except if someone beats me to it. Maybe @paglias will have an idea.

**Comment by lefnire** [comment]
@Fandekasp I honestly think both problems are an issue with jQuery UI Sortable. I've played around and noticed `from` & `to` getting funny values periodically. Give us a funny value where `to==list.length()`, and we'll get an extra tag. If I get time (could be a spell), I may try using a different sortable directive, like the one above

**Comment by lefnire** [comment]
Don't know how to fix the scrubbing issue, gotta be something in how we're styling our tags. The blank tags bit is due to edit & view -modes on tags being in the same `<ul>` and messing up jquery ui's indexing

File: public/js/directives/directives.js [hunk 1]
```diff
@@ -60,6 +60,24 @@ habitrpg.directive('habitrpgSortable', ['User', function(User) {
...
+  return function($scope, element, attrs, ngModel) {
+    var el = element;
+    $(element).sortable({
+      axis: "x",
+      start: function (event, ui) {
+        ui.item.data('startIndex', ui.item.index());
+      },
+      stop: function (event, ui) {
+        User.user.ops.sortTag({query:{
+            from: ui.item.data('startIndex') - 3,
```
Comment (lefnire) [inline public/js/directives/directives.js:73]: Incidentally, why -3?

Comment (AdrienLemaire) [inline public/js/directives/directives.js:73]: Otherwise the edit/hide/clear buttons also become sortable ^^

Comment (lefnire) [inline public/js/directives/directives.js:73]: I see. K, per https://github.com/HabitRPG/habitrpg/commit/50a34eb6c70cd8f042731a4d6925c023803de4dd#commitcomment-7448834 no longer necessary

**Pull request merged**